TFNetworkRecLayer.py

```diff
@@ -2838,33 +2872,23 @@ def get_loop_acc_layer(name):
       with tf.name_scope(self.layer_data_templates[name].layer_class_type.cls_get_tf_scope_name(name)):
         inner_layer = self.net.get_layer(name)
         acc_ta = loop_accumulated["output_%s" % name]
-        if inner_layer.get_search_choices() is None:
-          search_choices = None
-        else:
-          acc_ta, search_choices = self._opt_search_resolve(
-            layer_name=name, acc_ta=acc_ta, final_net_vars=final_net_vars)
+        acc_ta, search_choices, resolved_seq_len = self._opt_search_resolve(
+          layer_name=name, acc_ta=acc_ta, final_net_vars=final_net_vars, seq_len=seq_len)
```

Comment:
Why did you add the if statement here? Shouldn't `_opt_search_resolve` return `search_choices =None` and the original `acc_ta` in that case?

Reply:
No. In `_opt_search_resolve`, we use `layer_choice = self.net.get_search_choices(src=layer)`, which might not be the same, as I derived `LayerBase.get_search_choices()` for some layers. Although I think I undid that later. But maybe we can just use `layer_choice = layer. get_search_choices()` inside `_opt_search_resolve`.